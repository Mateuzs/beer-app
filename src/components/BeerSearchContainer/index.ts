export { default as BeerSearchContainer } from "./BeerSearchContainer";
