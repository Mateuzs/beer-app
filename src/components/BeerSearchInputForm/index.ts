export { default as BeerSearchInputForm } from "./BeerSearchInputForm";
