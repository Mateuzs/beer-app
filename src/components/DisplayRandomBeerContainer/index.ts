export { default as DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
