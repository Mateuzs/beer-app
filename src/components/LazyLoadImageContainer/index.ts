export { default as LazyLoadImageContainer } from "./LazyLoadImageContainer";
