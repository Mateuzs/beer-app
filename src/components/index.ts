export { BeerSearchContainer } from "./BeerSearchContainer";
export { DisplayRandomBeerContainer } from "./DisplayRandomBeerContainer";
export { NavBar } from "./NavBar";
export { SearchButton } from "./SearchButton";
export { LazyLoadImageContainer } from "./LazyLoadImageContainer";
export { ErrorMessage } from "./ErrorMessage";
export { BeerSearchInputForm } from "./BeerSearchInputForm";
export { BeerList } from "./BeerList";
