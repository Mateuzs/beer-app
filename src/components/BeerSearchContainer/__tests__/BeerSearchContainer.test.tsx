// external
import React from "react";
import { shallow, mount } from "enzyme";
// components, utils
import { BeerSearchContainer } from "../../../components";

describe("BeerSearchContainer component", () => {
  test("should match the snapshot and styles", () => {
    const mockedData = {
      isFetching: false,
      setIsFetching: jest.fn(),
      setIsError: jest.fn(),
    };
    const wrapper = shallow(<BeerSearchContainer {...mockedData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should render expected text", () => {
    const mockedData = {
      isFetching: false,
      setIsFetching: jest.fn(),
      setIsError: jest.fn(),
    };
    const component = mount(<BeerSearchContainer {...mockedData} />);

    expect(component.find('[data-test-id="beer-search-input-elements"]').length).toEqual(1);
    expect(component.find('[data-test-id="beer-search-input-search-button"]').length).toEqual(1);
    expect(component.find('[data-test-id="search-loading-info"]').length).toEqual(0);
  });
});
