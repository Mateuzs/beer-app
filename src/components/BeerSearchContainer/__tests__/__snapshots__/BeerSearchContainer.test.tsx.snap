// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerSearchContainer component should match the snapshot and styles 1`] = `
<div
  className="beer-search-container"
>
  <div
    className="beer-search-input-container"
  >
    <div
      className="beer-search-input-elements"
      data-test-id="beer-search-input-elements"
    >
      <Memo(BeerSearchInputForm)
        inputDescription="Only letters and numbers"
        inputPlaceholder="Name"
        inputValue=""
        isValidInputValue={true}
        onInputChangeCallback={[Function]}
        onInputRadioChangeCallback={[Function]}
      />
    </div>
    <div
      className="beer-search-input-search-button"
      data-test-id="beer-search-input-search-button"
    >
      <Memo(SearchButton)
        isDisabled={true}
        label="Search!"
        onClickHandler={[Function]}
      />
    </div>
  </div>
  <Memo(BeerList)
    beerList={Array []}
    isFetchedBeerList={false}
  />
</div>
`;
